src/components/Live/LiveProvider.js

```diff
@@ -55,7 +69,14 @@ class LiveProvider extends Component {
       scope
     }
     const errorCallback = err => this.setState({ element: undefined, error: err.toString() })
-    const renderElement = element => this.setState({ ...state, element })
+    const renderElement = element => this.setState({
+      ...state,
+      element: (
+        <ErrorBoundary handleError={errorCallback}>
```

Comment:
Can you move the `ErrorBoundary` code here: https://github.com/FormidableLabs/react-live/blob/master/src/utils/transpile/errorBoundary.js

You can also remove the code that is there right now, as it's just a limited try/catch for React 15. I had a more aggressive one there before, but it didn't work well.

Anyhow, it should probably live in that file :wink:

Reply:
Sure thing, do you have any preference on whether `ErrorBoundary` should be imported in this file and used to wrap `element`, or if `generateElement`/`renderElementAsync` should continue to return the element already wrapped in the boundary?

Reply:
The `errorBoundary.js` file is already used here: https://github.com/FormidableLabs/react-live/blob/master/src/utils/transpile/index.js

So basically, it's not like your code needs to be split up, it's just about its placement :wink: There's already that file which transforms the element that is being transpiled and spat out, so I guess we only need to replace the contents of `errorBoundary.js` by wrapping those contents with an actual React 16 Error boundary.